fontbakery-check-ttf.py

```diff
@@ -685,6 +720,149 @@ def ttf_file(f):
+    # to name IDs 1, 2, 4, 6, 16, 17, 18.
+    # It must run before any of the other name table related checks.
+
+    failed = False
+    # The font must have at least these name IDs:
+    for nameId in [NAMEID_FONT_FAMILY_NAME,
+                   NAMEID_FONT_SUBFAMILY_NAME,
+                   NAMEID_FULL_FONT_NAME,
+                   NAMEID_POSTSCRIPT_NAME,
+                   NAMEID_TYPOGRAPHIC_FAMILY_NAME,
+                   NAMEID_TYPOGRAPHIC_SUBFAMILY_NAME]:
+      if get_name_string(font, nameId) == None:
+        failed = True
+        fb.error(("Font lacks entry with"
+                  " nameId={} ({})").format(nameId,
+                                            NAMEID_STR[nameId]))
+
+    def family_with_spaces(value):
+      result = ''
+      for c in value:
+        if c.isupper():
+          result += " "
+        result += c
+      return result.strip()
+
+    def stylename_with_spaces(stylename):
+      if "Italic" in stylename:
+        return " Italic".join(stylename.split("Italic")).strip()
+      else:
+        return stylename
+
+    def get_only_weight(value):
+      onlyWeight = {"BlackItalic": "Black",
+                    "BoldItalic": "",
+                    "ExtraBold": "ExtraBold",
+                    "ExtraBoldItalic": "ExtraBold",
+                    "ExtraLightItalic": "ExtraLight",
+                    "LightItalic": "Light",
+                    "MediumItalic": "Medium",
+                    "SemiBoldItalic": "SemiBold",
+                    "ThinItalic": "Thin"}
+
+      if value in onlyWeight.keys():
+        return onlyWeight[value]
+      else:
+        return value
+
+    filename = os.path.split(font_file)[1]
+    filename_base = os.path.splitext(filename)[0]
+    fname, style = filename_base.split('-')
+    fname_with_spaces = family_with_spaces(fname)
+    style_with_spaces = stylename_with_spaces(style)
+    only_weight = get_only_weight(style)
+
+    for name in font['name'].names:
+      string = name.string.decode(name.getEncoding()).strip()
+      nameid = name.nameID
+      plat = name.platformID
+      expected_value = ""
+      expected_values = None
+
+      if nameid == NAMEID_FONT_FAMILY_NAME:
+        if plat == PLATFORM_ID_MACHINTOSH:
+          expected_value = fname_with_spaces
+        elif plat == PLATFORM_ID_WINDOWS:
+          if style in ['Regular',
+                       'Italic',
+                       'Bold',
+                       'Bold Italic']:
+            expected_value = fname_with_spaces
+          else:
+            expected_value = " ".join([fname_with_spaces,
+                                       only_weight]).strip()
+        else:
+          fb.error(("Font should not have a "
+                    "[{}:{}] entry!").format(NAMEID_STR[nameid],
+                                             PLATID_STR[plat]))
+          continue
+      elif nameid == NAMEID_FONT_SUBFAMILY_NAME:
+        if style_with_spaces not in STYLE_NAMES:
```

Comment:
This check is redundant with the checks done on the filename at the top of the script.